### DISCUSSION TITLE
API Rules - Blocking a specific field-change.

### DISCUSSION THREAD

**Discussion by bijsven** [pocketbase/pocketbase#4208] [Q&A] [answered]
Hey, is there a way to disable the update on a field via an API rule? Like I want to have some kind of permission system, so I dont want to have that people can update the "role" selector. Is that possible without using the Extend with code?

**Comment by ganigeorgiev** [comment, marked as answer]
You can use the [`:isset` field modifier](https://pocketbase.io/docs/api-rules-and-filters/#isset-modifier). For example:

```js
// either no role submitted or the submitted role matches with the existing value (aka. no change)
(@request.data.role:isset = false || role = @request.data.role)
```

**Reply by bijsven** [reply]
Works perfectly, ty!

**Comment by jonshipman** [comment]
Another solution could be to intercept onRecordBeforeUpdateRequest.

```
onRecordBeforeUpdateRequest((e) => {
  try {
    const requestInfo = $apis.requestInfo(e.httpContext);

    if (requestInfo.authRecord !== null) {
      e.record.set("role", "hardCodedRole");
    }
  } catch (e) {
    console.error(e);
  }
}, "collection_name");
```

The above example will always set collection_name.role to 'hardCodedRole'. You should put additional logic in there to determine the hardCodedRole based on your conditions. Like pulling in the previous record with `e.record.originalCopy()` and `.get('role')`. The check for authRecord is to prevent the block from executing on admins (though you can check for that specifically instead).
